File: source/old_cpp_src/_27_Remove_Element.c
int removeElement(int* nums, int numsSize, int val){
    int *p = nums;
    int *q = nums + numsSize - 1;

    while(p <= q)
    {
        // q指针从右到左查找要移除的元素
        if (*q == val)
        {
            /* code */
            q--;
            continue;
        }

        if (*p == val)
        {
            /* code */
            *p = *q;
            q--;
        }
        else
        {
            p++;
        }
        
    }
    return p - nums;
}

File: source/Go_LeetCode/27_RemoveElement.go
package Go_LeetCode

func removeElement(nums []int, val int) int {
	// 快慢指针
	slowIndex := 0
	for fastIndex := 0; fastIndex < len(nums); fastIndex++ {
		if nums[fastIndex] != val {
			nums[slowIndex] = nums[fastIndex]
			slowIndex++
		}
	}
	return slowIndex
}
